packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.test.ts

```diff
@@ -0,0 +1,1396 @@
+        },
+      ];
+      addUsers(users);
+
+      await prismock.membership.createMany({
+        data: [
+          {
+            id: 1,
+            teamId: teams[0].id,
+            userId: users[0].id,
+            role: MembershipRole.ADMIN,
+            accepted: true,
+          },
+          {
+            id: 2,
+            teamId: teams[0].id,
+            userId: users[1].id,
+            role: MembershipRole.MEMBER,
+            accepted: true,
+          },
+        ],
+      });
+
+      await prismock.outOfOfficeEntry.create({
+        data: {
+          userId: users[1].id,
+          uuid: uuidv4(),
+          start: new Date("2024-09-28T00:00:00.000Z"),
+          end: new Date("2024-10-03T23:59:59.999Z"),
+          notes: "",
+          reasonId: 1,
+          createdAt: new Date("2024-09-01T00:00:00.000Z"),
+          updatedAt: new Date("2024-09-01T00:00:00.000Z"),
+        },
+      });
+
+      const { schedules, ...loggedInUser } = users[0];
+      const ctx = {
+        user: loggedInUser as NonNullable<TrpcSessionUser>,
+      };
+
+      const input: TOutOfOfficeInputSchema = {
+        forUserId: users[1].id,
+        dateRange: {
+          startDate: new Date("2024-10-01T00:00:00.000Z"),
+          endDate: new Date("2024-10-01T00:00:00.000Z"),
+        },
+        offset: 330,
+        toTeamUserId: null,
+        reasonId: 1,
+      };
+
+      try {
+        await outOfOfficeCreateOrUpdate({ ctx, input });
+      } catch (e) {
+        expect(e).toBeInstanceOf(TRPCError);
+        expect((e as { code: string }).code).toEqual("CONFLICT");
+        expect((e as { message: string }).message).toBe("out_of_office_entry_already_exists");
+      }
+
+      //expect new outOfOffice Not to be created in db for userId:1
+      const ooo = await prismock.outOfOfficeEntry.findMany({
+        where: {
+          userId: users[1].id,
+        },
+      });
+      expect(ooo.length).toEqual(1);
+    });
+
+    it("should throw error if reverse redirect already exists, for team member.", async () => {
+      const teams = [
+        {
+          id: 1,
+          name: "Team 1",
+          slug: "team-1",
+        },
+      ];
+      await addTeamsToDb(teams);
+
+      const users = [
+        {
+          id: 1,
+          name: "User 1",
+          username: "user-1",
+          email: "user-1@test.com",
+          timeZone: "Asia/Kolkata",
+          schedules: [TestData.schedules.IstWorkHours],
+        },
+        {
+          id: 2,
+          name: "User 2",
+          username: "user-2",
+          email: "user-2@test.com",
+          timeZone: "Asia/Kolkata",
+          schedules: [TestData.schedules.IstWorkHours],
+        },
+        {
+          id: 3,
+          name: "User 3",
+          username: "user-3",
+          email: "user-3@test.com",
+          timeZone: "Asia/Kolkata",
+          schedules: [TestData.schedules.IstWorkHours],
+        },
+      ];
```

Comment:
As I mentioned, use a short example when a big list of items is not needed.

Reply:
Please refer reply to previous comment, here as well 3 users are required,
user[0] is admin,
user[1] is user for whom ooo is  created with redirect to user[2]